test/quicer_snb_SUITE.erl

```diff
@@ -846,11 +841,97 @@ tc_conn_no_gc(Config) ->
+                   , {stream_acceptors, 32}
+                   , {idle_timeout_ms, 1000}
+                   | default_conn_opts()],
+  StreamOpts = [ {stream_callback, quicer_echo_server_stream_callback}
+               | default_stream_opts() ],
+  Options = {ListenerOpts, ConnectionOpts, StreamOpts},
+  ct:pal("Listener Options: ~p", [Options]),
+  ?check_trace(#{timetrap => 20000},
+               begin
+                 %% Spawn a client process that will close the connection explicitly before die.
+                 %% The dead client process should trigger a connection close at server end.
+                 %% The dead client process should not trigger a GC of 'Conn' because the parent process
+                 %% still holds the 'Conn' var ref.
+                 {ok, _QuicApp} = quicer_start_listener(mqtt, Port, Options),
+                 Parent = self(),
+                 PRef = erlang:make_ref(),
+                 _Child = spawn_link(fun() ->
+                                         {ok, Conn} = quicer:connect("localhost", Port, [{idle_timeout_ms, 1000}, {alpn, ["sample"]}], 5000),
+                                         {ok, Stm} = quicer:start_stream(Conn, [{active, false}]),
+                                         Parent ! {PRef, Conn, Stm},
+                                         {ok, 4} = quicer:async_send(Stm, <<"ping">>),
+                                         {ok, <<"ping">>} = quicer:recv(Stm, 4),
+                                         quicer:shutdown_connection(Conn, 0, 0)
+                                     end),
+                 {ClientConn, ClientStream} = receive
+                                {PRef, C, S} -> {C, S}
+                              end,
+                 %% Server Process
+                 {ok, #{resource_id := _SRid}}
+                   = ?block_until(#{ ?snk_kind := debug
+                                   , context := "callback"
+                                   , function := "ServerConnectionCallback"
+                                   , mark := ?QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE
+                                   , tag := "event" },
+                                  5000, 1000),
+                 %% Client Process
+                 {ok, #{resource_id := CRid}}
+                   = ?block_until(#{ ?snk_kind := debug
+                                   , context := "callback"
+                                   , function := "ClientConnectionCallback"
+                                   , mark := ?QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE
+                                   , tag := "event" },
+                                  5000, 1000),
+                 %% Give it time for gc that should not happen on var 'Conn', could be the source of flakiness.
+                 %% We are rather testing the OTP behavior here but proves our understandings are correct.
+                 %% OTP GC callback, should not happen
```

Comment:
The test is we have two references of `Conn`, one in the client process spawned in L864 called `Conn`, another in this process L873 called `ClientConn` which is sent from client process.
calling `erlang:garbage_collect/0 ` will not triggered the GC of `ClientConn` in this process since it still hold the reference at L901.

The test is to verify the terminating of client process will not trigger resource dealloc callback of `Conn` since we still have a ref in `ClientConn`